src/transactions/TransactionFrame.cpp

```diff
@@ -1626,16 +1653,15 @@ TransactionFrame::applyOperations(SignatureChecker& signatureChecker,
             outerMeta.pushTxChangesAfter(std::move(changesAfter));
 
             if (protocolVersionStartsFrom(ledgerVersion,
-                                          SOROBAN_PROTOCOL_VERSION))
+                                          SOROBAN_PROTOCOL_VERSION) &&
+                isSoroban())
             {
-                if (!isSoroban() && !mEvents.empty())
-                {
-                    throw std::runtime_error("unexpected events size");
-                }
-
-                outerMeta.pushContractEvents(std::move(mEvents));
-                outerMeta.pushDiagnosticEvents(std::move(mDiagnosticEvents));
-                outerMeta.setReturnValue(std::move(mReturnValue));
+                outerMeta.pushContractEvents(
```

Comment:
@sisuresh I think that this is the correct change (we should never set "SorobanData" for non Soroban transactions)

Reply:
Yeah good catch. This didn't matter when these fields lived under `TransactionMetaV3`, but now that they're under the optional `SorobanTransactionMeta`, it should only be activated for Soroban.